conformance/tests/httproute-service-types.yaml

```diff
@@ -0,0 +1,122 @@
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: service-types
+  namespace: gateway-conformance-infra
+spec:
+  parentRefs:
+  - name: same-namespace
+  rules:
+  - matches: [{path: {type: Exact, value: /manual-endpoints}}]
+    backendRefs: [{name: manual-endpoints, port: 8080}]
+  - matches: [{path: {type: Exact, value: /manual-endpointslices}}]
+    backendRefs: [{name: manual-endpointslices, port: 8080}]
+  - matches: [{path: {type: Exact, value: /headless}}]
+    backendRefs: [{name: headless, port: 8080}]
+  - matches: [{path: {type: Exact, value: /headless-manual-endpoints}}]
+    backendRefs: [{name: headless-manual-endpoints, port: 8080}]
+  - matches: [{path: {type: Exact, value: /headless-manual-endpointslices}}]
+    backendRefs: [{name: headless-manual-endpointslices, port: 8080}]
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: manual-endpoints
+  namespace: gateway-conformance-infra
+spec:
+  ports:
+  - name: first-port
+    protocol: TCP
+    port: 8080
+    targetPort: 3000
+---
+apiVersion: v1
+kind: Endpoints
+metadata:
+  name: manual-endpoints
+  namespace: gateway-conformance-infra
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: manual-endpointslices
+  namespace: gateway-conformance-infra
+spec:
+  ports:
+  - name: first-port
+    protocol: TCP
+    port: 8080
+    targetPort: 3000
+---
+apiVersion: discovery.k8s.io/v1
+kind: EndpointSlice
+metadata:
+  name: manual-endpointslices
+  namespace: gateway-conformance-infra
+  labels:
+    kubernetes.io/service-name: manual-endpointslices
+addressType: IPv4
```

Comment:
This is probably not the safest assumption, you probably want to have EndpointSlices for both IP families and fill them with any addresses you find for either.

Reply:
This is a good catch - Is there a larger issue here about IPv6 support with Gateway API? That might warrant a separate conformance test/issue/discussion?

Reading up a bit I do see that `SingleStack` is the default for a K8s Services and the ip family used is the first family type from `service-cluster-ip-range` flag.

Reply:
We don't actually have any requirement in Gateway API that controllers support dual stack or a specific IP family, but I think our conformance tests should at least be possible to run in an IPv6-only cluster. As far as I know, this would be the first conformance test that would prevent that if we don't add EndpointSlices for both IP families.

Reply:
Added IPv6 support